lib/new_relic/agent/span_event_primitive.rb

```diff
@@ -80,6 +80,10 @@ def for_external_request_segment(segment)
           agent_attributes[HTTP_URL_KEY] = truncate(segment.uri)
         end
 
+        if segment.respond_to?(:record_agent_attributes?) && segment.record_agent_attributes?
+          agent_attributes.merge!(agent_attributes(segment))
+        end
+
```

Comment:
- Enhance external request segments and span event primitives so that the
  segments can now be set to record their agent attributes. When set to record
  them, the span event primitive will now merge them in along with the noticed
  errors and the http URL. Previously, only those 2 things wound up in the
  agent attributes and any other agent attributes belonging to the segment
  were ignored. The new `record_agent_attributes` writer and
  `record_agent_attributes?` check method are used in conjunction with a default
  of not recording agent attributes to make sure we're backwards compatible
  with the previous functionality of always ignoring agent attributes.